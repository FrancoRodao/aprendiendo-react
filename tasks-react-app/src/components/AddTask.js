import React from 'react'

export default function AddTask({ handleSubmit, handleInputChange, values }) {
    
    return (
        <>
            <form onSubmit={handleSubmit}>
                <p>Title: </p> <input onChange={handleInputChange} name="title" />
                <p>Desc: </p> <input onChange={handleInputChange} name="desc" />
                <p>Done: </p> <input onChange={handleInputChange} name="done" type="checkbox" />
                <button type="submit">Add task</button>
            </form>

            <div>
            <p>Preview: </p>
            <p>Title: </p> <input readOnly value={values.title} />
            <p>Desc: </p> <input readOnly value={values.desc} />
            <p>Done: </p> <input readOnly type="checkbox" checked={values.done} />
            </div>
        </>
    )
}
