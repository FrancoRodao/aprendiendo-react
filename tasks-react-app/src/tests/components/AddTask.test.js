import AddTask from "../../components/AddTask";
const { shallow } = require("enzyme")

describe('AddTask tests', () => {
    test('should display component correctly', () => {
        const formValues = {
            title: "task title",
            desc: "task desc",
            done: false
        }
        const wrapper = shallow(<AddTask values={formValues} />)
        expect(wrapper).toMatchSnapshot()
    })
    
})
