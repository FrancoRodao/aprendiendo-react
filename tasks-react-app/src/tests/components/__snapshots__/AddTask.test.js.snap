// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTask tests should display component correctly 1`] = `
<Fragment>
  <form>
    <p>
      Title: 
    </p>
     
    <input
      name="title"
    />
    <p>
      Desc: 
    </p>
     
    <input
      name="desc"
    />
    <p>
      Done: 
    </p>
     
    <input
      name="done"
      type="checkbox"
    />
    <button
      type="submit"
    >
      Add task
    </button>
  </form>
  <div>
    <p>
      Preview: 
    </p>
    <p>
      Title: 
    </p>
     
    <input
      readOnly={true}
      value="task title"
    />
    <p>
      Desc: 
    </p>
     
    <input
      readOnly={true}
      value="task desc"
    />
    <p>
      Done: 
    </p>
     
    <input
      checked={false}
      readOnly={true}
      type="checkbox"
    />
  </div>
</Fragment>
`;
